drivers/dai/intel/dmic/dmic.c

```diff
@@ -585,8 +571,7 @@ static void dai_dmic_start(struct dai_intel_dmic *dmic)
 
 		mic_a = dmic->enable[i] & 1;
 		mic_b = (dmic->enable[i] & 2) >> 1;
-		fir_a = (dmic->enable[i] > 0) ? 1 : 0;
-		fir_b = (dmic->enable[i] > 0) ? 1 : 0;
+		start_fir = !!(dmic->enable[i] > 0);
```

Comment:
Simplify !! ?

Reply:
Instead of two variables with the same value, we have one, so this is a simplification. In fact, double negation is unnecessary because the comparison returns a boolean value. I will fix it.